apps/o2-blocks/src/callout/editor.js

```diff
@@ -0,0 +1,349 @@
+import { __ } from '@wordpress/i18n';
+import { useMemo } from 'react';
+import metadata from './block.json';
+import {
+	CALLOUT_TEMPLATE_DEFAULT_COLOR,
+	CALLOUT_TEMPLATE_DEFAULT_BACKGROUND_COLOR,
+	CALLOUT_TEMPLATES,
+	CALLOUT_PANEL_COLORS,
+	CALLOUT_PANEL_BACKGROUND_COLORS,
+	CALLOUT_PANEL_DASHICONS,
+} from './constants';
+import { transforms } from './transform';
+
+import './editor.scss';
+import './view.scss';
+
+// === UTILITY FUNCTIONS ===
+
+const getIcon = ( templateName, customIcon = '' ) => {
+	if ( customIcon ) {
+		return customIcon.startsWith( 'dashicons-' ) ? customIcon : `dashicons-${ customIcon }`;
+	}
+
+	return CALLOUT_TEMPLATES[ templateName ]?.icon || CALLOUT_TEMPLATES.custom.icon;
+};
+
+const isValidColor = ( color ) => {
+	if ( ! color || typeof color !== 'string' ) {
+		return false;
+	}
+
+	return /^#[0-9A-F]{6}$/i.test( color );
+};
+
+const normalizeColor = ( color ) => {
+	if ( ! isValidColor( color ) ) {
+		return CALLOUT_TEMPLATE_DEFAULT_COLOR;
+	}
+
+	return color;
+};
+
+const normalizeBackgroundColor = ( color ) => {
+	if ( ! isValidColor( color ) ) {
+		return CALLOUT_TEMPLATE_DEFAULT_BACKGROUND_COLOR;
+	}
+
+	return color;
+};
+
+const getCurrentColor = ( templateName, customColor ) => {
+	return templateName === 'custom'
+		? normalizeColor( customColor )
+		: CALLOUT_TEMPLATES[ templateName ]?.color || CALLOUT_TEMPLATE_DEFAULT_COLOR;
+};
+
+// === COMPONENTS ===
+
+const CalloutIcon = ( { iconName, color } ) => (
+	<div className="o2-blocks-callout__icon" style={ { borderColor: color } }>
+		<span className={ `dashicons ${ iconName }` } style={ { color } } aria-hidden="true" />
+	</div>
+);
+
+const CalloutHeader = ( { content, onChange, placeholder, color, iconName } ) => (
+	<h2 className="o2-blocks-callout__header" style={ { color } }>
+		<CalloutIcon iconName={ iconName } color={ color } />
+		<span className="o2-blocks-callout__header-text">
+			<RichText
+				value={ content }
+				onChange={ onChange }
+				placeholder={ placeholder }
+				allowedFormats={ [ 'core/bold', 'core/italic' ] }
+				style={ { color } }
+			/>
+		</span>
+	</h2>
+);
+
+const IconPicker = ( { customIcon, currentColor, onIconChange, templateName } ) => {
+	const currentIconName = getIcon( templateName, customIcon );
+
+	const iconGrid = useMemo(
+		() =>
+			CALLOUT_PANEL_DASHICONS.map( ( iconName ) => {
+				const isSelected = currentIconName === iconName;
+
+				return (
+					<Button
+						key={ iconName }
+						onClick={ () => onIconChange( iconName ) }
+						variant="secondary"
+						className={ `o2-blocks-callout__icon-picker-button${
+							isSelected ? ' o2-blocks-callout__icon-picker-button--selected' : ''
+						}` }
+						style={ {
+							borderColor: isSelected ? currentColor : undefined,
+						} }
+						aria-label={ `Select ${ iconName.replace( 'dashicons-', '' ) } icon${
+							isSelected ? ' (currently selected)' : ''
+						}` }
+						title={ iconName }
```

Comment:
what is the reason for removing the `dashicons-` prefix on the `aria-label` but then leaving it in for the `title`? why did you make this choice?